### ISSUE TITLE
Request: Re-Categorize These to Relations 

### ISSUE THREAD

**Issue by TehTurk** [Frooxius/NeosPublic#1066]
Issue: Components that often use Relations or have the process of copying or similar in nature to ValueDriver and ValueMultiDriver should be within it. Alot are in Transform when they belong in a category that doesn't strictly deal with Transforms vs Deciding/Copying a value and how it relates. 

ValueCopy<T>
BooleanValueDriver<T>
DynamicBlendShapeDriver<T> [This would be better in Avatars as it's an Avatar Related Component or to be put in Rendering with it's Brother/Sister Components]
MultiBoolConditionDriver
ReferenceCopy<T>
ReferenceUserOverride<T>
StringConcatenationDriver<T>
TextureSizeDriver<T>
TimeIntDriver<T>
ValueEqualityDriver<T>
ValueGradientDriver<T>
ValueUserOverride<T>
ReferenceMultiplexer<T>
ValueMultiplexer<T>
MultiValueTextFormatDriver<T>

**Comment by Zyzyl** [comment]
While I agree that having e.g. ValueCopy(T) and ValueDriver(T) in different categories seems weird, I don't necessarily agree with this suggestion. If the goal is mainly to have ValueDriver and ValueMultiDriver in a place with more similar components, it would be a smaller change to simply move them to Transform/Drivers and get rid of the Relations category. That is, unless the intended purpose of Transform is to deal strictly with spatial transforms and its current contents doesn't correctly reflect that.

Can components be present in multiple categories? May not be the best solution though as I recall that idea was not well received when it was suggested for several LogiX nodes.

**Comment by TehTurk** [comment]
I would agree but this is also reducing the number of steps to get to these type of components as well. As contextually, Transforms would always deal with the position, rotation, or scale/transform of an object. Most of these while they can relate to transforms, don't always necessarily do and are more multi purpose and relationy with another Component oft not. This has been a consideration by a couple people I've talked to and catgorizing of some Components has been a thing for a bit.  

I mean you could? But then I don't know if it'd be clean/organized in that context.

**Comment by Frooxius** [comment]
We plan on recategorizing a lot of the components into proper categories. E.g. putting a lot of the ValueCopy and ValueDriver into Data and so on. However we've hold off the major recategorization for now to avoid confusing everyone's muscle memory all of sudden, until the component search is implemented. 

Do you feel it'd be warranted to just bite the bullet and recategorize them now or would waiting for the component search be better?

**Comment by Zyzyl** [comment]
In my opinion it would be better not to change a bunch of categories in the absence of larger changes (e.g. search) unless there is evidence (a vote, reactions to this issue...?) of significant support for doing so. I suspect it would annoy users whose workflows would be temporarily disrupted. While its true discoverability for many of these components isn't great for newer users, I'm not sure making this change right now would be particularly helpful to them either. Our current documentation (i.e. mostly ProbablePrime's videos) for some of the mentioned components refers to them in their current positions - a mismatch could cause confusion.

**Comment by TehTurk** [comment]
Hmm it depends, when would Component Search be added? This issue has been a bit of a backburner for me and I was just forgetting to make the actual issue. After last night I was trying to think of any leftover issues thst I hadn't made prior. Data would be more apt long as it's not 2 step, because oft not you know.you want ValCopy or BoolValDriver etc.

**Comment by Frooxius** [comment]
I don't know when exactly would Component Search be added. I want to rework the UI for it and make a lot of reusable pieces (e.g. folder tree browser UI), so it will be likely done in next batch of UI, but I'm not sure when that will come.

We can go ahead and recategorize everything at any point, it's relatively quick thing to do, but confusing users who are used to their existing locations is a concern. Search will alleviate that.

**Comment by TehTurk** [comment]
Hmm, I would say go ahead as there are other similar things that get added in this vein despite the slight annoyances so sooner would be better then later, but if there are others who wish for it to stay I do not mind. But can we please not put them in Data? Oft not they are Relation Components and are defined as such hence the suggestion for Relations.

**Comment by Frooxius** [comment]
I'd definitely want to gather more opinions on this and see if others wouldn't mind the disruption or breakage of existing tutorials.

Some of them belong in Data, since they deal with the data model and managing/relating data. They would likely have a subcategory there, the Relations category would just likely go away since it doesn't have a strong enough definition. Not all of them do though, so they will go to different places.

**Comment by H3BO3** [comment]
I'd rather not see tons of things moved around until native search functionality is added.

**Comment by TehTurk** [comment]
Thats a shame :c, Identity wise Relations would fit, as if say a Value is needed to be Driver and it's some sort of State Management, or a Way to copy, translate, or define data Relations would fit. Please don't seperate all of them into seperate little things as it makes it more harder then helpful. The reason I brought these up is also because they are the most.commonly/actively used Components. They should be readily accessible vs hidden under category under category.
